Index: backend/commons/model/src/main/java/app/model/cases/Application.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backend/commons/model/src/main/java/app/model/cases/Application.java b/backend/commons/model/src/main/java/app/model/cases/Application.java
new file mode 100644
--- /dev/null	(date 1638856049179)
+++ b/backend/commons/model/src/main/java/app/model/cases/Application.java	(date 1638856049179)
@@ -0,0 +1,45 @@
+package app.model.cases;
+
+import core.model.EntityBase;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.io.Serial;
+
+@Entity
+@Table(name = "application")
+@Getter
+@Setter
+public class Application extends EntityBase<Long> {
+
+    @Serial
+    private static final long serialVersionUID = 8760151928031866964L;
+
+    @Id
+    @Column(name = "id")
+    private Long id;
+
+    @NotNull
+    @ManyToOne
+    @JoinColumn(name = "case_id", updatable = false, nullable = false)
+    private CaseEntity caseEntity;
+
+    @Column(name = "agent_id")
+    private Integer agentId;
+
+    @Column(name = "tpl_id")
+    private Integer tplId;
+
+    @Column(name = "agent_vud_id")
+    private Integer agentVudId;
+
+    @Column(name = "tpl_vud_id")
+    private Integer tplVudId;
+}
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	(revision 5258b0c285ee5832f00c0ae61f8c212235afa76a)
+++ b/.idea/misc.xml	(date 1639559945247)
@@ -2,8 +2,8 @@
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
   <component name="FrameworkDetectionExcludesConfiguration">
-    <file type="web" url="file:///mnt/e590e9d2-31d9-4dee-b286-43a91e05069e/work/hunter-app" />
     <file type="web" url="file://$PROJECT_DIR$" />
+    <file type="web" url="file:///mnt/e590e9d2-31d9-4dee-b286-43a91e05069e/work/hunter-app" />
   </component>
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
@@ -52,5 +52,5 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="17" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_16" project-jdk-name="17" project-jdk-type="JavaSDK" />
 </project>
\ No newline at end of file
Index: backend/commons/model/src/main/java/app/model/cases/CaseEntity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backend/commons/model/src/main/java/app/model/cases/CaseEntity.java b/backend/commons/model/src/main/java/app/model/cases/CaseEntity.java
new file mode 100644
--- /dev/null	(date 1638856775162)
+++ b/backend/commons/model/src/main/java/app/model/cases/CaseEntity.java	(date 1638856775162)
@@ -0,0 +1,71 @@
+package app.model.cases;
+
+import core.model.EntityBase;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.io.Serial;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Entity
+@Table(name = "cases")
+@Getter
+@Setter
+public class CaseEntity extends EntityBase<Long> {
+
+    @Serial
+    private static final long serialVersionUID = 3645589819205373375L;
+
+    @Id
+    @Column(name = "id")
+    private Long id;
+
+    @NotNull
+    @Column(name = "client_id", nullable = false, updatable = false)
+    private Integer clientId;
+
+    @Column(name = "date_start")
+    private LocalDateTime dateStart;
+
+    @Column(name = "date_end")
+    private LocalDateTime dateEnd;
+
+    @Column(name = "weight_sum")
+    private Boolean weightSum;
+
+    @Column(name = "app_count")
+    private int appCount;
+
+    @Column(name = "priority")
+    private Integer priority;
+
+    @Column(name = "role_id")
+    private Integer roleId;
+
+    @Size(max = 256)
+    @Column(name = "post_region")
+    private String postRegion;
+
+    @Size(max = 256)
+    @Column(name = "full_name")
+    private String fullName;
+
+    @OneToMany(mappedBy = "caseEntity")
+    private List<Application> applicationList = new java.util.ArrayList<>();
+
+    @ManyToMany
+    @JoinTable(name = "case_triggers", joinColumns = @JoinColumn(name = "case_id"),
+            inverseJoinColumns = @JoinColumn(name = "trigger_id"))
+    private List<Trigger> triggers;
+}
Index: backend/commons/model/src/main/java/app/model/cases/CaseQueue.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backend/commons/model/src/main/java/app/model/cases/CaseQueue.java b/backend/commons/model/src/main/java/app/model/cases/CaseQueue.java
new file mode 100644
--- /dev/null	(date 1638859901665)
+++ b/backend/commons/model/src/main/java/app/model/cases/CaseQueue.java	(date 1638859901665)
@@ -0,0 +1,30 @@
+package app.model.cases;
+
+import core.model.EntityBase;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import java.io.Serial;
+
+@Entity
+@Getter
+@Setter
+public class CaseQueue extends EntityBase<CaseEntity> {
+
+    @Serial
+    private static final long serialVersionUID = 3560073564655375282L;
+
+    @Id
+    @OneToOne
+    @JoinColumn(name = "id")
+    private CaseEntity id;
+
+    @Column(name = "afc_priority")
+    private Integer afcPriority;
+
+}
Index: backend/commons/model/src/main/java/app/model/cases/Trigger.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backend/commons/model/src/main/java/app/model/cases/Trigger.java b/backend/commons/model/src/main/java/app/model/cases/Trigger.java
new file mode 100644
--- /dev/null	(date 1638856095245)
+++ b/backend/commons/model/src/main/java/app/model/cases/Trigger.java	(date 1638856095245)
@@ -0,0 +1,23 @@
+package app.model.cases;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.io.Serial;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "triggers")
+public class Trigger implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 8193076595624017881L;
+
+    @Id
+    @Column(name = "id")
+    private Integer id;
+
+    @Column(name = "name")
+    private String name;
+}
Index: .idea/dataSources.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/dataSources.xml b/.idea/dataSources.xml
--- a/.idea/dataSources.xml	(revision 5258b0c285ee5832f00c0ae61f8c212235afa76a)
+++ b/.idea/dataSources.xml	(date 1638611461920)
@@ -5,7 +5,7 @@
       <driver-ref>h2.unified</driver-ref>
       <synchronize>true</synchronize>
       <jdbc-driver>org.h2.Driver</jdbc-driver>
-      <jdbc-url>jdbc:h2:$PROJECT_DIR$/backend/test-db/build/resources/main/test</jdbc-url>
+      <jdbc-url>jdbc:h2:file:$PROJECT_DIR$/backend/test-db/build/resources/main/test</jdbc-url>
       <working-dir>$ProjectFileDir$</working-dir>
     </data-source>
     <data-source source="LOCAL" name="HUNTER - DEV2" uuid="89f7a9a7-5b80-4660-8aae-644fcd7e6aba">
Index: .jpb/jpb-settings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.jpb/jpb-settings.xml b/.jpb/jpb-settings.xml
new file mode 100644
--- /dev/null	(date 1638612253073)
+++ b/.jpb/jpb-settings.xml	(date 1638612253073)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="JpaPluginProjectSettings">
+    <option name="lastSelectedLanguage" value="Java" />
+  </component>
+</project>
\ No newline at end of file
Index: backend/db/src/main/resources/db-migration/main/0.0.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backend/db/src/main/resources/db-migration/main/0.0.xml b/backend/db/src/main/resources/db-migration/main/0.0.xml
--- a/backend/db/src/main/resources/db-migration/main/0.0.xml	(revision 5258b0c285ee5832f00c0ae61f8c212235afa76a)
+++ b/backend/db/src/main/resources/db-migration/main/0.0.xml	(date 1638860023984)
@@ -224,4 +224,104 @@
         </rollback>
     </changeSet>
 
+    <changeSet id="0021" author="lesin-sa">
+        <createTable tableName="cases">
+            <column name="id" type="bigint">
+                <constraints primaryKey="true" primaryKeyName="pk_cases"/>
+            </column>
+            <column name="created_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="updated_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="client_id" type="int">
+                <constraints nullable="false"/>
+            </column>
+            <column name="date_start" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="date_end" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="weight_sum" type="boolean"/>
+            <column name="app_count" type="int">
+                <constraints nullable="false"/>
+            </column>
+            <column name="priority" type="int"/>
+            <column name="role_id" type="int"/>
+            <column name="post_region" type="varchar(256)"/>
+            <column name="full_name" type="varchar(256)"/>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="0022" author="lesin-sa">
+        <createTable tableName="application">
+            <column name="id" type="bigint">
+                <constraints primaryKey="true" primaryKeyName="pk_application"/>
+            </column>
+            <column name="created_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="updated_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="case_id" type="bigint">
+                <constraints nullable="false" foreignKeyName="fk_application_cases" referencedTableName="application"
+                             referencedColumnNames="id"/>
+            </column>
+            <column name="agent_id" type="int"/>
+            <column name="tpl_id" type="int"/>
+            <column name="agent_vud_id" type="int"/>
+            <column name="tpl_vud_id" type="int"/>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="0023" author="lesin-sa">
+        <createTable tableName="triggers">
+            <column name="id" type="int">
+                <constraints primaryKey="true" primaryKeyName="pk_triggers"/>
+            </column>
+            <column name="name" type="varchar(127)">
+                <constraints nullable="false" unique="true" uniqueConstraintName="uq_triggers"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="0024" author="lesin-sa">
+        <createTable tableName="case_triggers">
+            <column name="case_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="trigger_id" type="int">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="0025" author="lesin-sa">
+        <addPrimaryKey constraintName="pk_case_triggers" tableName="case_triggers" columnNames="case_id,trigger_id"/>
+        <rollback>
+            <dropPrimaryKey tableName="case_triggers" constraintName="pk_case_triggers"/>
+        </rollback>
+        <comment>pk_case_triggers</comment>
+    </changeSet>
+
+    <changeSet id="0026" author="lesin-sa">
+        <createTable tableName="case_queue">
+            <column name="id" type="bigint">
+                <constraints primaryKey="true" primaryKeyName="pk_case_queue" foreignKeyName="fk_case_queue_id"
+                             referencedTableName="cases" referencedColumnNames="id" deleteCascade="true"/>
+            </column>
+            <column name="created_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="updated_at" type="datetime">
+                <constraints nullable="false"/>
+            </column>
+            <column name="afc_priority" type="int">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
 </databaseChangeLog>
